### PR TITLE
V3 content

### PR DESCRIPTION
**Pull request by itowlson**
This branch tracks work on Spin 3 updates. Initially it's a clone of the Spin 2 docs but this is where we will add / remove / tweak for the v3 stuff.

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [ ] The `title`, `template`, and `date` are all set
- [ ] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [ ] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep $'\r' | wc -l` and expect 0 as a result)
- [ ] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [ ] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [ ] Headings are using Title Case
- [ ] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [ ] Have tested with `npm run test` and resolved all errors
- [ ] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

**Comment by itowlson** [comment]
@karthik2804 I had to update some logic in `page_lang.hbs` and `spin_main.hbs` to tell it to use the v3 sidebar, but I couldn't figure out how to do the `if ... else if ... else ...` required to handle v3, v2 _and_ v1.  Can you advise please?  (See https://github.com/fermyon/developer/pull/1392/commits/8dd50f8c532f707a7274d715c32fddbfd1108a1c for the crimes.)

**Comment by karthik2804** [comment]
@itowlson I opened a PR for that.

**Comment by itowlson** [comment]
Is there a way to know why preview deployment is failing?  All the logs say is Internal Server Error.

**Comment by vdice** [comment]
Are we continuing the tradition of a cli reference?  If so, do we want to add to this PR?  Or a follow-up?  There is a toolkit that should help with scaffolding it out: https://github.com/fermyon/developer/tree/main/toolkit

**Comment by itowlson** [comment]
No, I want to try getting rid of the CLI reference in its current format because it's laborious to maintain and doesn't seem to add much value.  I'd maybe like to add a page that lists the commands and links to the other docs pages that cover them, which will be more stable and I feel more useful.  But I can land that in a later PR.

File: content/spin/v3/rdbms-storage.md
Comment (vdice) [inline content/spin/v3/rdbms-storage.md:1]: Any updates needed to this doc re: https://github.com/fermyon/spin/pull/2890?

**Review by vdice — COMMENTED** [review]
Mostly double-checking if there any applicable docs needing updates for 3.0.  Tracking any/all in follow-ups works with me -- especially if we could use other folks to contribute.

File: content/spin/v3/extending-and-embedding.md
Comment (itowlson) [reply content/spin/v3/extending-and-embedding.md:1]: Good call.  There have been some shenanigans here.  Let's raise a new issue for this so as not to block the application developer v3 content.

File: content/spin/v3/kv-store-api-guide.md
Comment (itowlson) [reply content/spin/v3/kv-store-api-guide.md:1]: The existing stuff is still true.  We can do this once we know what the SDK side of things looks like.

File: content/spin/v3/rdbms-storage.md
Comment (itowlson) [reply content/spin/v3/rdbms-storage.md:1]: Can we hold off until SDK support takes shape?

File: content/spin/v3/extending-and-embedding.md
Comment (vdice) [reply content/spin/v3/extending-and-embedding.md:1]: 👍 https://github.com/fermyon/developer/issues/1411

**Review by vdice — APPROVED** [review]
LGTM pending actual v3 release :)

**Review by macolso — APPROVED** [review]
Looks good to me. I left a few small suggestions that shouldn't be considered blockers. 

Will Fermyon Cloud support 3.0? Either way, I think it would be good to update Cloud documentation accordingly (e.g., adding component dependencies to the [Spin Limitations section](https://fermyon-developer-pr-1392-xcy9wthx.fermyon.app/cloud/faq#spin-limitations))

File: content/spin/v3/writing-apps.md
Comment (itowlson) [reply content/spin/v3/writing-apps.md:441]: That could be useful, although maybe it needs to be less experimental and more documented?  I worry that sending users there right now would raise more questions than it answers.  But yeah definitely worth keeping in mind.

File: content/spin/v3/writing-apps.md
Comment (itowlson) [reply content/spin/v3/writing-apps.md:513]: 'Main' was intended to refer to the component that the dependency is a dependency of.  Like the 'main' one in the, uh, component (composition), not the 'main' one in the application.  I'll try to rephrase but would welcome ideas!

Comment (itowlson) [reply content/spin/v3/writing-apps.md:513]: Okay I think this gets clearer if I talk about the 'component manifest' rather than a 'main' or 'parent' component e.g.

> If a component has a dependency which requires resource access, you can grant it by setting the `dependencies_inherit_configuration` flag in the Spin component manifest:
> ...
> This grants _all_ dependencies access to _all_ resources listed in the Spin component manifest. You should therefore set this only if you trust _all_ dependencies.

File: content/spin/v3/migration-v2-v3.md
Comment (vdice) [reply content/spin/v3/migration-v2-v3.md:10]: It will be https://github.com/fermyon/spin/releases/v3.0.0

File: content/cloud/faq.md
Comment (michelleN) [inline content/cloud/faq.md:89]: Should we add a link to wasi cloud core? https://github.com/WebAssembly/wasi-cloud-core

File: content/cloud/faq.md
Comment (michelleN) [inline content/cloud/faq.md:91]:
```suggestion
| [Component dependencies](https://github.com/fermyon/spin/blob/main/docs/content/sips/020-component-dependencies.md) | Not supported |
```

File: content/spin/v3/contributing-spin.md
Comment (michelleN) [inline content/spin/v3/contributing-spin.md:34]:
```suggestion
First, any contribution and interaction on the Spin project MUST follow our
```

File: content/spin/v3/contributing-spin.md
Comment (michelleN) [inline content/spin/v3/contributing-spin.md:35]:
```suggestion
[code of conduct](https://github.com/fermyon/spin/blob/main/CODE_OF_CONDUCT.md). Thank you for being
```

File: content/cloud/faq.md
Comment (itowlson) [reply content/cloud/faq.md:91]: Linking to the docs rather than the SIP but good call 👍

File: content/spin/v3/writing-apps.md
Comment (karthik2804) [reply content/spin/v3/writing-apps.md:441]: I would be in favour of waiting for the plugin to become less experimental as Ivan suggests.

File: content/spin/v3/writing-apps.md
Comment (macolso) [reply content/spin/v3/writing-apps.md]: I like that approach!

**Pull request merged**
